quiz/views.py

```diff
@@ -0,0 +1,109 @@
+            care_drinker = all_answer[6][0]
+            goals        = all_answer[7]
+            vegan_type   = all_answer[8][0]
+            allergies    = all_answer[9]
+            diseases     = all_answer[10]
+
+            gender_code     = [1, 3] if gender == 'male' else [2, 3]
+            age_code        = [2, 3] if age >= 50 else [1, 3]
+            bmi             = weight/((height*0.01)**2)
+            care_obesity    = [0, 1] if  bmi>= 25 else [0]
+            activity_level  = [3] if activity == 'workoutMore6hours' else [1,2]
+            care_smoker     = bool(care_smoker == 'yesSmoke')
+            care_drinker    = [0, 1] if care_drinker == 'drinkingMore' else [0]
+            goal_id_list    = [goal.id for goal in Goal.objects.filter(name__in=goals)]
+            vegan_level     = {vegan_type == 'vegan': [1], vegan_type == 'vegetarian': [1, 2]}.get(True, [1, 2 ,3])
+            allergy_id_list = [allergy.id for allergy in Allergy.objects.filter(name__in=allergies)]
+            disease_id_list = [disease.id for disease in Disease.objects.filter(name__in=diseases)]
+            
+            q = Q()
+            q.add(Q(gender_code__in = gender_code) & Q(age_level__in = age_code) 
+                & Q(care_obesity__in = care_obesity) & Q(activity_level__in = activity_level) 
+                & Q(care_smoker = care_smoker) & Q(care_drinker__in = care_drinker) 
+                & Q(goal__in = goal_id_list) & Q(vegan_level__in = vegan_level), q.OR)
+
+            if allergy_id_list:
+                q.add(~Q(allergy__in = allergy_id_list), q.AND)
+
+            if disease_id_list:
+                q.add(Q(disease__in = disease_id_list), q.AND)
+
+            results_products = Product.objects.filter(q | Q(is_default=True)).distinct()
+                      
+            if QuizResult.objects.filter(user=user):
+                user.recommendation.clear()
+            
+            recommendations = []
+            for result_product in results_products:
+                user.recommendation.add(result_product)
+
+                product_SSPs   = result_product.productstock_set.all()
+                price_list     = [product_SSP.price for product_SSP in product_SSPs] #product_price_list
+                size_list      = [product_SSP.size for product_SSP in product_SSPs]
+                goals          = result_product.goal.all()
+                goal_name_list = [goal.name for goal in goals]
+
+                product_info = {}
+                product_info["id"]         = result_product.id
+                product_info["title"]      = result_product.name
+                product_info["subTitle"]   = result_product.sub_name
+                product_info["imageUrl"]   = result_product.image_set.get(is_main=True).image_url
+                product_info["healthGoal"] = goal_name_list
+                product_info["price"]      = price_list
+                product_info["size"]       = size_list
+
```

Comment:
```suggestion

                product_info = {}
                product_info["id"]         = result_product.id
                product_info["title"]      = result_product.name
                product_info["subTitle"]   = result_product.sub_name
                product_info["imageUrl"]   = result_product.image_set.get(is_main=True).image_url
                product_info["healthGoal"] = goal_name_list
                product_info["price"]      = price_list
                product_info["size"]       = size_list

```
`product_info`를 여러번 타이핑 하셨을 것 같은데 
딕셔너리를 선언함과 동시에 key-value를 입력하는 방법을 추천드립니다.
`product_info`라는 변수가 너무 많이 나타나고, 키값이 한 번에 눈에 들어오는 형태가 아니기 때문입니다 :)

Reply:
네넹~